provider/bidengine/order.go

```diff
@@ -182,8 +194,31 @@ loop:
 			}
 
 			if bidFound {
-				o.bidPlaced = true
-				o.session.Log().Info("Found existing bid ")
+				o.session.Log().Info("found existing bid")
+				bidResponse := queryBid.Value().(*mtypes.QueryBidResponse)
+				bid := bidResponse.GetBid()
+				bidState := bid.GetState()
+				if bidState != mtypes.BidOpen {
+					o.session.Log().Error("bid in unexpected state", "bid-state", bidState)
+					break loop
+				}
+				bidPlaced = true
+
+				if o.bidTimeoutEnabled() {
+					// This bid could be very old, compute the minimum age of the bid
+					// do not try anything clever here like asking the RPC node for the current height
+					// just use the height from when the session is created
+					createdAtBlock := bid.GetCreatedAt()
+					blockAge := createdAtBlock - o.session.CreatedAtBlockHeight()
+					const minTimePerBlock = 5 * time.Second
+					atLeastThisOld := time.Duration(blockAge) * minTimePerBlock
+					if atLeastThisOld > o.cfg.BidTimeout {
+						o.session.Log().Info("found expired bid", "block-height", createdAtBlock)
+						break loop
+					}
+				}
```

Comment:
put this in a method or function `isStaleBid()`

Reply:
This is done now, it wraps up nicely into a helper